cmd/controller/main.go

```diff
@@ -53,108 +52,39 @@ type controller struct {
 	clientset kubernetes.Interface
 }
 
-func readKey(client kubernetes.Interface, namespace, keyName string) (*rsa.PrivateKey, []*x509.Certificate, error) {
-	secret, err := client.Core().Secrets(namespace).Get(keyName, metav1.GetOptions{})
-	if err != nil {
-		return nil, nil, err
-	}
-
-	key, err := certUtil.ParsePrivateKeyPEM(secret.Data[v1.TLSPrivateKeyKey])
-	if err != nil {
-		return nil, nil, err
+func initNames(prefix, label *string) (string, string, error) {
```

Comment:
I'm confused by this function ;)
The name is very generic, it takes references to values (which it modifies), and then it also returns values? 

Afaics, all it _actually_ does is check the characters on each value, and maybe append a `-` to one of them.  I think I would prefer we just fix the default values (to include the `-` suffix or not), and then let the k8s server return an error if the user gave us something with the wrong characters.

Reply:
Fair enough, I've simplified it to just validation and added the default value.